pybamm/discretisations/discretisation.py

```diff
@@ -53,7 +53,10 @@ def process_model(self, model):
 
         """
         # set boundary conditions (only need key ids for boundary_conditions)
-        self._bcs = {key.id: value for key, value in model.boundary_conditions.items()}
+        self._bcs = {
+            key.id: self.process_dict(value)
+            for key, value in model.boundary_conditions.items()
+        }
```

Comment:
for clarity the boundary conditions should be processed outside of this function (and maybe `self._bcs` should be changed to `self._processed_bcs`)